creme/creme_core/views/enumerable.py

```diff
@@ -20,19 +20,52 @@
 from django.http import Http404
 from django.shortcuts import get_object_or_404
 
+from creme.creme_core.enumerators import CustomFieldEnumerator
+
 from ..core.enumerable import enumerable_registry
 from ..core.exceptions import BadRequestError, ConflictError
 from ..http import CremeJsonResponse
-from ..models import CustomField, CustomFieldEnumValue
+from ..models import CustomField
 from .generic import base
 
 
 class ChoicesView(base.ContentTypeRelatedMixin, base.CheckedView):
     response_class = CremeJsonResponse
-    field_url_kwarg = 'field'
-    registry = enumerable_registry
     limit_arg = 'limit'
     term_arg = 'term'
+    only_arg = 'only'
+
+    def get_enumerator(self):
+        raise NotImplementedError()
+
+    def read_limit_arg(self, request):
+        try:
+            limit = request.GET.get(self.limit_arg)
+            return int(limit) if limit is not None else None
+        except (TypeError, ValueError) as e:
+            raise BadRequestError(e) from e
+
+    def read_only_arg(self, request):
```

Comment:
`def get_limit(self)` & `def get_only(self)` to be consistent with the other views.
+ you continue to put to many code in try..except.. block.